codalab/lib/bundle_cli.py

```diff
@@ -47,60 +50,158 @@
 from codalab.machines.local_machine import LocalMachine
 from codalab.client.remote_bundle_client import RemoteBundleClient
 from codalab.lib.formatting import contents_str
+from codalab.lib.completers import CodaLabCompleter, WorksheetsCompleter
+
+
+## Formatting Constants
+GLOBAL_SPEC_FORMAT = "[<alias>::|<address>::]|(<uuid>|<name>)"
+ADDRESS_SPEC_FORMAT = "(<alias>|<address>)"
+TARGET_SPEC_FORMAT = '[<key>:](<uuid>|<name>)[%s<subpath within bundle>]' % (os.sep,)
+BUNDLE_SPEC_FORMAT = '(<uuid>|<name>|^<index>)'
+WORKSHEET_SPEC_FORMAT = GLOBAL_SPEC_FORMAT
+GROUP_SPEC_FORMAT = '(<uuid>|<name>|public)'
+PERMISSION_SPEC_FORMAT = '((n)one|(r)ead|(a)ll)'
+UUID_POST_FUNC = '[0:8]'  # Only keep first 8 characters
+
+class Commands(object):
+    '''
```

Comment:
Reading this is a bit abstract, can you give an example of how Argument and Command should be used?